itests/org.openhab.core.ephemeris.tests/src/main/java/org/eclipse/smarthome/core/ephemeris/internal/EphemerisManagerImplOSGiTest.java

```diff
@@ -233,4 +236,81 @@ public void testConfigOptionProviderCitiesTasmania() {
         assertFalse(options.isEmpty());
         assertEquals(ephemerisManager.cities.get(REGION_TASMANIA_KEY), options);
     }
+
+    @Test
+    public void testWeekends() {
+        ZonedDateTime monday = ZonedDateTime.of(2019, 10, 28, 0, 0, 0, 0, ZoneId.of("Europe/Paris"));
+        ZonedDateTime sunday = ZonedDateTime.of(2019, 10, 27, 0, 0, 0, 0, ZoneId.of("Europe/Paris"));
+        ephemerisManager.modified(Collections.singletonMap(CONFIG_DAYSET_PREFIX + INTERNAL_DAYSET,
+                Stream.of("Monday", "Tuesday").collect(Collectors.toList())));
+        assertEquals(true, ephemerisManager.isWeekend(sunday));
+        assertEquals(false, ephemerisManager.isWeekend(monday));
+        assertEquals(true, ephemerisManager.isInDayset(INTERNAL_DAYSET, monday));
+        assertEquals(false, ephemerisManager.isInDayset(INTERNAL_DAYSET, sunday));
+    }
+
+    @Test
+    public void testIsBankHoliday() {
+        ZonedDateTime newyearsday = ZonedDateTime.of(2019, 01, 01, 0, 0, 0, 0, ZoneId.of("Europe/Paris"));
+        ZonedDateTime secondday = ZonedDateTime.of(2019, 01, 02, 0, 0, 0, 0, ZoneId.of("Europe/Paris"));
+
+        boolean vacation = ephemerisManager.isBankHoliday(newyearsday);
+        assertEquals(true, vacation);
+
+        vacation = ephemerisManager.isBankHoliday(secondday);
+        assertEquals(false, vacation);
+    }
+
+    @Test
+    public void testGetBankHoliday() {
+        ZonedDateTime theDay = ZonedDateTime.of(2019, 01, 01, 0, 0, 0, 0, ZoneId.of("Europe/Paris"));
+
+        boolean vacation = ephemerisManager.isBankHoliday(theDay);
+        assertEquals(true, vacation);
+
+        String code = ephemerisManager.getBankHolidayName(theDay);
+        assertEquals("NEW_YEAR", code);
+
+        String name = ephemerisManager.getHolidayDescription(code);
```

Comment:
Travis failed because of this test. I guess you have to apply the French locale in advance:

```
ephemerisManager.modified(Collections.singletonMap(CONFIG_COUNTRY, Locale.FRANCE.getCountry()));
```

Reply:
Modified but Travis seems to complain on Java 11. Should I try to address this ?

Reply:
I am afraid the failures are related to #1170 (see also https://github.com/openhab/openhab2-addons/issues/6334).